packages/flutter/lib/src/gestures/tap.dart

```diff
@@ -62,6 +62,7 @@ class TapUpDetails {
   TapUpDetails({
     this.globalPosition = Offset.zero,
     Offset localPosition,
+    this.kind,
```

Comment:
if you mark this @required, does it catch any cases where we're not setting it? we should make sure to set it everywhere, even if we don't end up marking it actually @required.

Reply:
I already went through all the usages and made sure to add the `kind` argument. The only exception is the one generated from semantics, in which case there's no device kind associated with it. Should I just use `touch` as the default?

Reply:
Oh there's a `PointerDeviceKind.unknown`. I think it makes more sense to use this instead of defaulting to touch.